packages/button/src/Button.ts

```diff
@@ -127,11 +144,25 @@ export class Button extends SizedMixin(ButtonBase, { noDefaultSize: true }) {
     public set quiet(quiet: boolean) {
         this.treatment = quiet ? 'outline' : 'fill';
     }
-
     public get quiet(): boolean {
         return this.treatment === 'outline';
     }
 
+    protected override willUpdate(
+        changedProperties: PropertyValues<this>
+    ): void {
+        super.willUpdate(changedProperties);
+        if (changedProperties.has('pending')) {
+            if (this.pending) {
+                if (!this.disabled) {
+                    this.setAttribute('aria-label', this.pendingLabel);
+                }
```

Comment:
Good move here. We do need coverage for when `this.disabled` moves to `false` and `this.pending` is already `true` to ensure the `aria-label` is appropriately applied in that case.